src/executor/stack/executor.rs

```diff
@@ -821,6 +833,30 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		}
 	}
 
+	/// Calculate gas limit and record it in the gasometer.
+	fn calc_gas_limit_and_record(
+		&mut self,
+		target_gas: Option<u64>,
+		take_l64: bool,
+	) -> Result<u64, ExitError> {
+		let after_gas = if take_l64 && self.config.call_l64_after_gas {
```

Comment:
You can declare a variable which would allow you to write the function shorter: 
```rust
let gasometer = &mut self.state.metadata_mut().gasometer;
```

Reply:
You propose declaring mutability also for immutable cases. It's a bad idea.

Reply:
456d5218fbdd653a9bc3d1a9a4cb83a039d7f814